GitUI/Editor/FileViewer.cs

```diff
@@ -1557,18 +1548,33 @@ private void ApplySelectedLines(bool allFile, bool reverse)
                 "--whitespace=nowarn"
             };
 
-            string output = Module.GitExecutable.GetOutput(args, patch, throwOnErrorOutput: false);
-            ProcessApplyOutput(output, patch);
+            ProcessApplyOutput(args, patch);
         }
 
-        private void ProcessApplyOutput(string output, byte[] patch, bool patchUpdateDiff = false)
+        private void ProcessApplyOutput(GitArgumentBuilder args, byte[] patch, bool patchUpdateDiff = false)
         {
-            if (!string.IsNullOrEmpty(output))
+            ExecutionResult result = Module.GitExecutable.Execute(args, inputWriter => inputWriter.BaseStream.Write(patch), throwOnErrorOutput: false);
+            string output = result.AllOutput.Trim();
+            if (EnvUtils.RunningOnWindows())
             {
-                if (patchUpdateDiff || !MergeConflictHandler.HandleMergeConflicts(UICommands, this, false, false))
-                {
-                    MessageBox.Show(this, output + "\n\n" + Encoding.GetString(patch), TranslatedStrings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                }
+                // remove file mode warnings
+                Regex regEx = new("warning: .*has type .* expected .*");
+                output = output.RemoveLines(regEx.IsMatch);
+            }
+
+            if (!result.ExitedSuccessfully && (patchUpdateDiff || !MergeConflictHandler.HandleMergeConflicts(UICommands, this, false, false)))
+            {
+                MessageBox.Show(this, $"{output}\n\n{Encoding.GetString(patch)}", TranslatedStrings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            else if (!result.ExitedSuccessfully || output.StartsWith("error: ") || output.StartsWith("warning: "))
+            {
+                // git-apply may fail on first attempt but succeed in subequent attempts
+                // Trace such occurrences that may be interesting, some of these should maybe be presented to the user
+                Trace.WriteLineIf(!string.IsNullOrWhiteSpace(output), $"Patch output: {result.ExitCode}:{output} for: git {args}");
+            }
+            else if (!string.IsNullOrWhiteSpace(output))
+            {
+                Debug.WriteLine($"Patch output: {result.ExitCode}:{output} for: git {args}");
```

Comment:
Just curious: How do see this output between all the noisy "thread has exited" messages?

Reply:
We could add the [error] output to the Git Command Log.

Reply:
> Just curious: How do see this output between all the noisy "thread has exited" messages?

Right now, you dont unless you break in the debugger or view a log.
There is a need to set trace levels, to view some messages only.

> We could add the [error] output to the Git Command Log.

I have also considered that, for messages the user could be interested in (not thread exit or time to parse git-log for revGrid...)

For Debug vs Trace, I would handle the output the same, just not visible in Release.

Reply:
What's the point of these trace and debug messages? These aren't visible to a user...

`(!result.ExitedSuccessfully || output.StartsWith("error: ") || output.StartsWith("warning: "))` to me implies there was an error - why are we not showing an error/throwing?

Reply:
The approach to ignoring popups is quite aggressive here. This can be motivated with that there are no subsequent GE processing.
However, if this is too aggressive it is possible to debug this with something like https://docs.microsoft.com/en-us/sysinternals/downloads/debugview
(or edit .ini to print to console)

I would like to clean up the trace printouts and set levels that can be configured.

> to me implies there was an error - why are we not showing an error/throwing?

Git has succeeded to apply the patch with a secondary strategy, so it *should* not be an error